client/analytics/report/products/table.js

```diff
@@ -0,0 +1,227 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { map, orderBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { Card, Link, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { downloadCSVFile, generateCSVDataFromTable, generateCSVFileName } from 'lib/csv';
+import { formatCurrency, getCurrencyFormatDecimal } from 'lib/currency';
+import { onQueryChange } from 'lib/nav-utils';
+import ReportError from 'analytics/components/report-error';
+
+export default class ProductsReportTable extends Component {
+	constructor( props ) {
+		super( props );
+	}
+
+	onDownload( headers, rows, query ) {
+		// @TODO The current implementation only downloads the contents displayed in the table.
+		// Another solution is required when the data set is larger (see #311).
+		return () => {
+			downloadCSVFile(
+				generateCSVFileName( 'products', query ),
+				generateCSVDataFromTable( headers, rows )
+			);
+		};
+	}
+
+	getHeadersContent() {
+		return [
+			{
+				label: __( 'Product Title', 'wc-admin' ),
+				key: 'name',
+				required: true,
+				isLeftAligned: true,
+			},
+			{
+				label: __( 'SKU', 'wc-admin' ),
+				key: 'sku',
+				hiddenByDefault: true,
+			},
+			{
+				label: __( 'Items Sold', 'wc-admin' ),
+				key: 'items_sold',
+				required: true,
+				defaultSort: true,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'G. Revenue', 'wc-admin' ),
+				key: 'gross_revenue',
+				required: true,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Orders', 'wc-admin' ),
+				key: 'orders_count',
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Category', 'wc-admin' ),
+				key: 'product_cat',
+			},
+			{
+				label: __( 'Variations', 'wc-admin' ),
+				key: 'variation',
+			},
+			{
+				label: __( 'Status', 'wc-admin' ),
+				key: 'stock_status',
+			},
+			{
+				label: __( 'Stock', 'wc-admin' ),
+				key: 'stock',
+			},
+		];
+	}
+
+	getRowsContent( data = [] ) {
+		const { stockStatuses } = wcSettings;
+
+		return map( data, row => {
+			const {
+				product_id,
+				sku = '', // @TODO
+				name,
+				items_sold,
+				gross_revenue,
+				orders_count,
+				categories = [], // @TODO
+				variations = [], // @TODO
+				stock_status = 'outofstock', // @TODO
+				stock_quantity = '0', // @TODO
+			} = row;
+
+			return [
+				{
+					// @TODO Must link to the product detail report.
+					display: name,
+					value: name,
+				},
+				{
+					display: sku,
+					value: sku,
+				},
+				{
+					display: items_sold,
+					value: items_sold,
+				},
+				{
+					display: formatCurrency( gross_revenue ),
+					value: getCurrencyFormatDecimal( gross_revenue ),
+				},
+				{
+					display: (
+						<Link
+							href={
+								'admin.php?page=wc-admin#/analytics/orders?filter=advanced&product_includes=' +
```

Comment:
Super cool :star:

Reply:
Also should we pull in the current query args here? I.E. if I'm viewing this report for last quarter, if I click on this link I would expect it to pull up the same date filters.

Reply:
Good idea here. I had the need to pull out time related queries once before and am thinking it would be a good abstraction. Maybe in `/components/filters/utils`?

https://github.com/woocommerce/wc-admin/blob/9c199dec7cf60b23d5552b8bcb6b55fd80dc957a/client/components/filters/filter/index.js#L106

Reply:
Also, if you use `type="wc-admin"`, the `href` attribute can simply be `analytics/orders?filter=advanced&product_includes=...`

Reply:
> Also should we pull in the current query args here? I.E. if I'm viewing this report for last quarter, if I click on this link I would expect it to pull up the same date filters.

Right, actually I was waiting for #568 to be merged so I could reuse some of the work done there. Do you think that would make sense @psealock?